tools/witx/src/io.rs

```diff
@@ -0,0 +1,88 @@
+use crate::WitxError;
+use std::collections::HashMap;
+use std::fs::{read_to_string, File};
+use std::io::{BufRead, BufReader, Error, ErrorKind};
+use std::path::{Path, PathBuf};
+
+pub trait WitxIo {
+    fn fgets(&self, path: &Path) -> Result<String, WitxError>;
```

Comment:
This would benefit from a doc comment, along the lines of "Read the whole contents of a file into a string".